tutorials/getting-started/reusable-content-with-pieces.md

```diff
@@ -0,0 +1,335 @@
+      name: 'body',
+      label: 'Biography',
+      type: 'area',
+      options: {
+        widgets: {
+          'apostrophe-rich-text': {
+            controls: [ 'Bold', 'Italic', 'Link', 'Unlink' ]
+          },
+          'apostrophe-images': {}
+        }
+      }
+    },
+    {
+      name: 'phone',
+      label: 'Phone',
+      type: 'string'
+    },
+    {
+      name: 'thumbnail',
+      label: 'Thumbnail',
+      type: 'singleton',
+      widgetType: 'apostrophe-images',
+      options: {
+        limit: 1,
+        minSize: [ 800, 800 ],
+        aspectRatio: [ 1, 1 ]
+      }
+    }
+  ]
+};
+```
+
+> **IMPORTANT: note the `name` property. This identifies ONE piece in the database, so it is always singular.** Remember: Modules Are Plural (MAP), but the things they manage may not be.
+
+Now let's turn the module on in `app.js`. *From now on, we'll show `modules` with just the modules we're adding. Of course you will have other modules in your `app.js` file as well.*
+
+```javascript
+modules: {
+  // ...,
+  'people': {}
+}
+```
+
+Just like that, you'll see a new "People" menu in the admin bar when you log into the site. Pick "New Person" and you'll find that you can give each person a full name, a first name, a last name and a phone number. Pick "Manage Person" to examine and edit existing people.
+
+> We could have configured the module entirely in `app.js`. But that leads to giant `app.js` files, so we don't recommend it. However, some developers feel it's a good place for high-level properties like `extend` that help give you a quick overview of what the module is and does.
+
+There are certain additional fields that you get by default with every piece, such as `title` (the full name of the piece), `slug` (used when the piece appears as part of a URL), and `published` (which determines whether the public can see the piece, as you'll see below). But in this case, we re-declared `title` in order to change its label to `Full Name` so that the "New Person" form is not confusing.
+
+You can even add a profile photo, via the `thumbnail` field. This field has the `singleton` type, which allows us to include a widget in the schema for this type of piece, exactly as if we were calling `apos.singleton` in a template. We just need to specify the `widgetType` and pass any desired options to the widget via the `options` property. You can also add fields of the `area` type.
+
+And, there's a "biography" section. This is a full-blown content area in which the editor can add rich text and images. *There's nothing to stop us from allowing more controls and widgets here. Limiting the choices just helps keep things from getting out of hand.*
+
+### Customizing the model layer: setting the `title` automatically
+
+Right now, the `title` property (which is always the full name of the piece) is independent of `firstName` and `lastName`. For people, it makes more sense for the `title` to be generated automatically from `firstName` and `lastName`.
```

Comment:
One thing to note on this is that `slug` also generates from `title` by default, at least on the frontend, so making the `title` field contextual might be a little weird, since you'll have to hand-pick a slug. Is this alright? @boutell